core/src/test/scala/unit/kafka/log/LogCleanerTest.scala

```diff
@@ -834,10 +835,10 @@ class LogCleanerTest {
     val initialLogSize = log.size
 
     val (endOffset, stats) = cleaner.clean(LogToClean(new TopicPartition("test", 0), log, 2, log.activeSegment.baseOffset))
-    assertEquals(5, endOffset)
+    assertEquals(3, endOffset)
```

Comment:
@hachikuji I had to change these values for them to pass. In my mind, they make sense. Can you confirm they're correct?

Reply:
I think this is something to do with the loadFactor passed to `makeCleaner`. A value of 2 is only putting one record in the offsetMap. With a value of 4, we get `maxDesiredMapSize` of 3. And we check `map.size < maxDesiredMapSize`, so that allows for 2 values to be put in map, which is same as number of unique records to compact.

I tried a value of 4 and test passes w/o any other change.

Reply:
I think this is something to do with the loadFactor passed to makeCleaner. A value of 2 is only putting one record in the offsetMap. With a value of 4, we get maxDesiredMapSize of 3. And we check map.size < maxDesiredMapSize, so that allows for 2 values to be put in map, which is same as number of unique records to compact.

I tried a value of 4 and test passes w/o any other change.